src/rbd.cc

```diff
@@ -2470,6 +2469,34 @@ static int do_show_status(librados::IoCtx &io_ctx, librbd::Image &image,
   return 0;
 }
 
+static int do_rename(librbd::RBD &rbd, librados::IoCtx& io_ctx,
+		     librbd::Image &image, const char *imgname, const char *destname)
+{
+  int r = rbd.open_read_only(io_ctx, image, imgname, NULL);
+  if (r < 0) {
+    cerr << "rbd: error opening image " << imgname << ": "
+         << cpp_strerror(-r) << std::endl;
+    return -r;
+  }
+  bool watch_exists = false;
+  r = do_show_status(io_ctx, image, imgname, NULL, &watch_exists);
+  if (r < 0)
+    return r;
+  if (watch_exists) {
+    cerr << "rbd: watchers on this image, unmap it first" << std::endl;
+    return -EBUSY;
+  }
+  r = image.close();
```

Comment:
nevermind, it does make sense to open and close it locally in this function, since format 1 images would be problematic. No need to pass in the image parameter from main() though.